File: src/include/creator/CreatorLicenseType.php
<?php

namespace Creator;

enum CreatorLicenseType: int
{
	case ANY_LICENSE = 2;
	case PUBLIC_DOMAIN = 1;


	public function title(): string
	{
		return match ($this) {
			self::PUBLIC_DOMAIN => 'Public Domain Only',
			self::ANY_LICENSE => 'Any License',
		};
	}

	public function slug(): string
	{
		return match ($this) {
			self::PUBLIC_DOMAIN => 'public-domain',
			self::ANY_LICENSE => 'any',
		};
	}

	public static function tryFromSlug(string $slug): ?self
	{
		return match ($slug) {
			'public-domain' => self::PUBLIC_DOMAIN,
			'any-license' => self::ANY_LICENSE,
			default => null,
		};
	}
}

File: app/creator/creator_license_type.py
from enum import IntEnum
from typing import Optional


class CreatorLicenseType(IntEnum):
    PUBLIC_DOMAIN = 1
    OPEN_LICENSE = 2
    ANY_LICENSE = 3

    def title(self) -> str:
        return {
            CreatorLicenseType.PUBLIC_DOMAIN: "Public Domain Only",
            CreatorLicenseType.OPEN_LICENSE: "Open License",
            CreatorLicenseType.ANY_LICENSE: "Any License",
        }[self]

    def slug(self) -> str:
        return {
            CreatorLicenseType.PUBLIC_DOMAIN: "public-domain",
            CreatorLicenseType.OPEN_LICENSE: "open",
            CreatorLicenseType.ANY_LICENSE: "any",
        }[self]

    @classmethod
    def from_slug(cls, slug: str) -> "CreatorLicenseType":
        return cls.try_from_slug(slug) or cls.ANY_LICENSE

    @classmethod
    def try_from_slug(cls, slug: str) -> Optional["CreatorLicenseType"]:
        for member in cls:
            if member.slug() == slug:
                return member
        return None
